### PR TITLE
Accordion icon style fix

### LINKED ISSUE #23163: [Bug]: Accordion: Expand icon does not rotate on iPad Safari
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
iPad Pro (12.9in)
iPadOS 15.4.1
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://react.fluentui.dev/?path=/docs/components-accordion--default#default

## Actual Behavior

When tapping an Accordion header the expand icon does not change to indicate the header is open. 

## Expected Behavior

The rotation of the Accordion header expand icon should change to indicate if the header is expanded or collapsed.

This is probably because the rotation is handled as an attribute on the SVG rather than as a CSS style: https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeader.tsx#L73

This implementation also prevents [custom `expandIcon`s](https://react.fluentui.dev/?path=/docs/components-accordion--default#expand-icon) from rotating as the rotation is only applied to the default icon.

The first screen in the video is the iPad, the second screen in Edge 101.0.1210.53 on Windows 11

https://user-images.githubusercontent.com/93940821/169924005-ccdf6b60-5e92-4221-aeac-abd7d32f43c7.MOV

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by tomi-msft**
When tapping the expand icon in Safari, the icon doesn't change to indicate that it is open.  This is because Safari does not support the `transform` attribute on an svg currently.

This PR refactors the styling in order for it to be applied in Safari

Fixes #23163 and #24035

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                               Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |  `79.349 kB`<br />`24.053 kB` |  `79.357 kB`<br />`24.057 kB` | `8 B` <br />`4 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `188.818 kB`<br />`51.901 kB` | `188.826 kB`<br />`51.905 kB` | `8 B` <br />`4 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `33.259 kB`<br />`10.952 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d1e8f0764fb477271da45f0dd1f4c1b3beb06699'>d1e8f0764fb477271da45f0dd1f4c1b3beb06699</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/Avatar-mount_base.html">1315</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/Avatar-mount.html">1311</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/Button-mount_base.html">952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/Button-mount.html">964</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/FluentProvider-mount_base.html">1546</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/FluentProvider-mount.html">1539</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/FluentProviderWithTheme-mount_base.html">639</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/FluentProviderWithTheme-mount.html">619</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">577</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">605</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">626</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">633</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/MakeStyles-mount_base.html">1803</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/MakeStyles-mount.html">1791</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/SpinButton-mount_base.html">2443</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24597/perf-test-react-components/SpinButton-mount.html">2530</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by spmonahan — APPROVED** [review]

File: packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeader.tsx [hunk 1]
```diff
@@ -67,7 +67,7 @@ export const useAccordionHeader_unstable = (
     expandIcon: resolveShorthand(expandIcon, {
       required: true,
       defaultProps: {
-        children: <ChevronRightRegular transform={`rotate(${expandIconRotation})`} />,
+        children: <ChevronRightRegular style={{ transform: `rotate(${expandIconRotation}deg)` }} />,
```
Comment (micahgodbolt) [inline packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeader.tsx:70]: my only thought here is, how would someone else customize the expand icon, and re-use this rotation?

Comment (tomi-msft) [reply packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeader.tsx:70]: The guidance to users adding styling in our docs is to use `makeStyles` and add it to the `className` so I think that approach would still work here

Comment (micahgodbolt) [reply packages/react-components/react-accordion/src/components/AccordionHeader/useAccordionHeader.tsx:70]: are they able to alter the className based on the open/closed state? A demo would probably be helpful to show others how we expect them to do it. Rotate an icon, change + to -, 'open' to 'closed' etc

**Review by micahgodbolt — APPROVED** [review]

**Pull request merged**